### ISSUE TITLE
[Bug]: When trying to create a Java String for a enum value a `TypeCastError` is thrown

### ISSUE THREAD

**Issue by ayeshLK** [ballerina-platform/ballerina-lang#42829]
### Description

> $subject

### Steps to Reproduce

Consider the following code segment:

```ballerina
import ballerina/jballerina.java as java;

public enum LogLevel {
    DEBUG,
    INFO,
    WARNING,
    ERROR,
    FATAL,
    OFF
}

public type CustomConfiguration record {
    LogLevel logLevel = OFF;
};

configurable CustomConfiguration customConfiguration = {};

public isolated client class SampleClient {
    private final readonly & LogLevel logLevel;

    public isolated function init() returns error? {
        self.logLevel = customConfiguration.logLevel;
        handle logHandle = java:fromString(self.logLevel);
    }
}

public function main() returns error? {
    SampleClient clientEp = check new;
}
```

Following runtime error is thrown when trying to run the above program:
```
error: {ballerina}TypeCastError {"message":"incompatible types: '()' cannot be cast to 'string'"}
	at ayesh.asb_issue_test.0.SampleClient:init(client.bal:25)
	   ayesh.asb_issue_test.0:main(client.bal:31)
```

### Affected Version(s)

Ballerina SL 2201.9.0

### Related area

-> Runtime

### Related issue(s) (optional)
- [#6573](https://github.com/ballerina-platform/ballerina-library/issues/6573)

**Comment by HindujaB** [comment]
It seems like the Immutable type definition we create for configurable variable types does not pass the correct fields to type nodes at the code analyzing phase. This behavior is there for the previous versions as well (`2201.8.x` and before). Due to the record default value creation changes in 2201.9.0, it is now resulting in an error. 
<img width="1427" alt="Screenshot 2024-06-03 at 13 22 49" src="https://github.com/ballerina-platform/ballerina-lang/assets/28644893/9b73e502-dc92-4ca5-823d-7270e2594792">

The following is the simplified sample.
```ballerina
import ballerina/io;
public type CustomConfiguration record {
    string logLevel = "OFF";
};

configurable CustomConfiguration customConfiguration = {};

public function main() returns error? {
    io:println(customConfiguration);
    CustomConfiguration c = {}; // prints {"logLevel":null} should be {"logLevel":"OFF"}
}
```

**Comment by chiranSachintha** [comment]
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/42853

**Comment by ayeshLK** [comment]
We are facing a similar issue in the daily build of Ballerina Azure service bus package [1]. In the daily build we are using the ballerina-nightly docker image. I checked with ballerina-2201.9.2 docker image and it was working for me locally. But, when switched to ballerina-nightly docker image this was reproducible locally. 

Hence, reopening this issue.

@chiranSachintha / @anuruddhal FYI


[1] - https://github.com/ballerina-platform/module-ballerinax-azure-service-bus/actions/runs/10001697992/job/27645785481

**Comment by MaryamZi** [comment]
Looks like the ballerina-nightly docker image may not be up to date. Last push for ballerina:nightly seems to be 2 months ago - https://hub.docker.com/r/ballerina/ballerina/tags. Matches the digest in the log too. cc @anuruddhal @keizer619 

@ayeshLK, I think we can use a separate issue to track this, since the original issue seems to have been fixed on 2201.9.2 and https://github.com/ballerina-platform/ballerina-lang/pull/42852 was merged to master.

**Comment by keizer619** [comment]
> Looks like the ballerina-nightly docker image may not be up to date. Last push for ballerina:nightly seems to be 2 months ago - https://hub.docker.com/r/ballerina/ballerina/tags. Matches the digest in the log too. cc @anuruddhal @keizer619
> 
> @ayeshLK, I think we can use a separate issue to track this, since the original issue seems to have been fixed on 2201.9.2 and #42852 was merged to master.

There are test failures in daily build which causes docker not to push we are looking into that

**Comment by ayeshLK** [comment]
Created another issue [1] to track this. Will close this issue as this has been already fixed.

[1] - https://github.com/ballerina-platform/module-ballerina-c2c/issues/799
